src/components/TextField/TextField.scss

```diff
@@ -106,6 +113,10 @@ $stacking-order: (
   font-weight: inherit;
   appearance: none;
 
+  .hasValue & {
+    color: var(--p-text-on-surface, color('black'));
```

Comment:
There appears to be a bug in Safari when using `inherit` as a `var` fallback (`var(--p-text-on-surface, inherit)`) will inherit a light ink color instead of the user agent stylesheet which sets input text color to black.

Reply:
So strange. Good catch @tmlayton!